File: moviejava/app/src/main/java/com/example/moviejava/ui/home/fragment/genres/GenresAdapter.java
package com.example.moviejava.ui.home.fragment.genres;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.moviejava.databinding.ItemGenresBinding;
import com.example.moviejava.extentions.OnClickItem;
import com.example.moviejava.model.Genres;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;




public class GenresAdapter extends RecyclerView.Adapter<GenresAdapter.ViewHolder> {
    @Inject
    public GenresAdapter() {
    }

    private final List<Genres> listGenres= new ArrayList<>();
    public OnClickItem<Genres> onClickItem;

    @SuppressLint("NotifyDataSetChanged")
    public void submitData(List<Genres> data) {
        listGenres.clear();
        listGenres.addAll(data);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(
                ItemGenresBinding.inflate(
                        LayoutInflater.from(parent.getContext()),
                        parent,
                        false
                )
        );
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.onBind(listGenres.get(position));
        holder.itemView.setOnClickListener(view -> onClickItem.onClick(listGenres.get(position)));
    }

    @Override
    public int getItemCount() {
        return listGenres.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        private final ItemGenresBinding binding;

        public ViewHolder(@NonNull ItemGenresBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void onBind(Genres g) {
            binding.setG(g);
            binding.executePendingBindings();
        }
    }
}

File: movie/app/src/main/java/com/example/movie/ui/home/fragment/genres/GenresAdapter.kt
package com.example.movie.ui.home.fragment.genres

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.movie.databinding.ItemGenresBinding
import com.example.movie.model.Genres
import javax.inject.Inject


class GenresAdapter @Inject constructor() : RecyclerView.Adapter<GenresAdapter.ViewHolder>() {
    private var listGenres = mutableListOf<Genres>()
    var onClickItem: ((Genres) -> Unit)? = null
    fun submitData(data: List<Genres>) {
        listGenres.clear()
        listGenres.addAll(data)
        notifyDataSetChanged()
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            ItemGenresBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        listGenres[position].let {
            holder.onBind(it)
            holder.itemView.setOnClickListener { _ ->
                onClickItem?.invoke(it)
            }
        }
    }

    override fun getItemCount() = listGenres.size

    inner class ViewHolder(private val binding: ItemGenresBinding) :
        RecyclerView.ViewHolder(binding.root) {
        fun onBind(genres: Genres) {
            binding.g = genres
            binding.executePendingBindings()
        }
    }
}